src/main/scala/calc/Compiler.scala

```diff
@@ -61,21 +62,113 @@ object Compiler {
     ir.Hashers.hashClassDef(classDef)
   }
 
+
+
+  def findCapture(tree: Tree): Set[String] = {
+    tree match {
+      case _:Literal => Set()
+      case BinaryOp(_, l, r) => findCapture(l) ++ findCapture(r)
+      case Let(i, v, body) => findCapture(body) ++ findCapture(v) - i.name
+      case Ident(n) => Set(n)
+      case If(cond, thenp, elsep) => findCapture(cond) ++ findCapture(thenp) ++ findCapture(elsep)
+      case Closure(idents, body) => findCapture(body) -- idents.flatMap((x:Tree) => findCapture(x)).toSet
+      case Call(f, args) => findCapture(f) ++ args.flatMap((x:Tree) => findCapture(x)).toSet
+    }
+  }
+
+
+  def compileExpr(tree: Tree):irt.Tree ={
+    implicit val pos = tree.pos
+    typeCheck(tree)
+    implicit val mathModule = irt.LoadModule(irtpe.ClassType("jl_Math$"))
```

Comment:
It would be much better to create this tree where you need it, when you need it. For starters, you wouldn't have to thread it around the entire code. But also it would have the appropriate `pos` at each place where it is needed.

Also, it makes more sense, independently of this assignment. In a real compiler, you could not afford passing around such hard-coded trees for all sorts of stuff you might need everywhere.